Format.pm

```diff
@@ -505,24 +536,50 @@ variables, use C<format_number()> instead.
 
 sub round
 {
-    my ($self, $number, $precision) = _get_self @_;
+    my ($self, $number, $precision, $roundoption) = _get_self @_;
 
     unless (defined($number))
     {
         _complain_undef();
         $number = 0;
     }
 
-    $precision = $self->{decimal_digits} unless defined $precision;
-    $precision = 2 unless defined $precision;
+    $precision   = $self->{decimal_digits} unless defined $precision;
+    $precision   = 2 unless defined $precision;
+    $roundoption = $self->{round_option} unless defined $roundoption;
+    $roundoption = ROUND_NORMAL unless defined $roundoption;
 
     croak("precision must be integer")
         unless int($precision) == $precision;
 
     if (ref($number) && $number->isa("Math::BigFloat"))
     {
         my $rounded = $number->copy();
-        $rounded->precision(-$precision);
+
```

Comment:
OK, for ROUND_ABS_FLOOR and ROUND_ABS_CEIL it is possible to call is_neg() twice. But to set $sign correctly we need to call is_neg() and is_pos() to get $sign = -1|0|1 (like $sign at other parts in the code).


```
            my $sign = $rounded->is_neg() ? -1
                     : $rounded->is_pos() ?  1
                     :                       0;
```

Additional i think the options ROUND_ABS_FLOOR & ROUND_ABS_CEIL are rarely used, so better call is_neg() twice when this options are set then call is_neg() every time without the need (for ROUND_FLOOR, ROUND_NORMAL and ROUND_CEIL).